File: 2920/2920.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    2920.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2023/05/07 21:07:49 by jihoolee          #+#    #+#              #
#    Updated: 2023/05/07 21:13:07 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main():
    arr = map(int, input().split())
    sorted_arr = sorted(arr)

    if (arr == sorted_arr):
        print('ascending')
    elif (arr == sorted_arr[::-1]):
        print('descending')
    else:
        print('mixed')


if (__name__ == '__main__'):
    main()

File: 2920/2920.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   2920.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/05/07 19:05:13 by jihoolee          #+#    #+#             */
/*   Updated: 2023/05/07 21:07:01 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

const int LEN = 8;

int main(void) {
  std::vector<int>  arr(LEN);
  int               asc = 0;
  int               desc = 0;

  for (size_t idx = 0; idx < LEN; ++idx)
    std::cin >> arr[idx];
  for (size_t idx = 0; idx < LEN / 2; ++idx) {
    if (arr[idx] + arr[LEN - 1 - idx] != LEN + 1)
      break;
    else if (arr[idx] < arr[idx + 1])
      ++asc;
    else if (arr[idx] > arr[idx + 1])
      ++desc;
  }
  if (asc == 4)
    std::cout << "ascending";
  else if (desc == 4)
    std::cout << "descending";
  else
    std::cout << "mixed";
  return 0;
}
